ng/manage.ts

```diff
@@ -436,6 +436,72 @@ angular.module("armExplorer", ["ngRoute", "ngAnimate", "ngSanitize", "ui.bootstr
         window.location.href = "api/tenants/" + $scope.selectedTenant.id;
     };
 
+    $scope.resourceSearch = () => {
+        // trigger a refresh if needed, but do not need to wait for it to finish since cache should already exist
+        refreshResourceSearchCache();
+
+        var results: IResourceSearchSuggestion[] = [];
+        var keyword = $scope.resourceSearchModel.searchKeyword || "";
+        var stringCompare = (strA: string, strB: string): number=> {
+            if (strA > strB) {
+                return 1;
+            } else if (strA < strB) {
+                return -1;
+            } else {
+                return 0;
+            }
+        };
+
+        $scope.resourceSearchCache.data.forEach((item, index) => {
```

Comment:
you can use [`Array.filter()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) rather than a `forEach` I think it'll be clearer what you're doing here